zerver/worker/queue_processors.py

```diff
@@ -1139,6 +1140,17 @@ def failure_processor(event: Dict[str, Any]) -> None:
             )
             user_profile = get_user_profile_by_id(event["user_profile_id"])
             reactivate_user_if_soft_deactivated(user_profile)
+        elif event["type"] == "scrub_deactivated_realms":
+            realm_to_scrub = Realm.objects.get(id=event["realm_id"])
+
+            if (
+                realm_to_scrub.scheduled_deletion_date is not None
+                and realm_to_scrub.scheduled_deletion_date <= timezone_now()
+            ):
+                do_scrub_realm(realm_to_scrub, acting_user=None)
+                realm_to_scrub.scheduled_deletion_date = None
+                realm_to_scrub.save()
+                logger.info("Scrubbed realm %s", realm_to_scrub.id)
```

Comment:
What's the thinking on this going into a queue? I don't see a reason the cron job couldn't process things directly.

Reply:
Yeah, it was done for the `immediate` option to work.